src/common/entity/compute_attribute_display.ts

```diff
@@ -1,34 +1,116 @@
 import { HassEntity } from "home-assistant-js-websocket";
+import { html, TemplateResult } from "lit";
+import { until } from "lit/directives/until";
 import { EntityRegistryDisplayEntry } from "../../data/entity_registry";
 import { HomeAssistant } from "../../types";
+import checkValidDate from "../datetime/check_valid_date";
+import { formatDate } from "../datetime/format_date";
+import { formatDateTimeWithSeconds } from "../datetime/format_date_time";
+import { formatNumber } from "../number/format_number";
+import { capitalizeFirstLetter } from "../string/capitalize-first-letter";
+import { isDate } from "../string/is_date";
+import { isTimestamp } from "../string/is_timestamp";
 import { LocalizeFunc } from "../translations/localize";
 import { computeDomain } from "./compute_domain";
 
+let jsYamlPromise: Promise<typeof import("../../resources/js-yaml-dump")>;
+
 export const computeAttributeValueDisplay = (
-  localize: LocalizeFunc,
-  stateObj: HassEntity,
-  entities: HomeAssistant["entities"],
+  hass: HomeAssistant,
+  entity: HassEntity | string,
   attribute: string,
   value?: any
-): string => {
-  const entityId = stateObj.entity_id;
-  const deviceClass = stateObj.attributes.device_class;
+): string | TemplateResult => {
```

Comment:
@bramkragten, should we pass `localize` and `entities` instead of `hass` object? If we use `lit-context` we need to pass individual item instead of the full hass object.

Reply:
Don't forget about `hass.local`, which is now used for number formatting.

Reply:
Yes, it's easier to pass `hass` object. 
We discussed in front-end meeting to avoid passing the whole `hass` object to every component in the future to use `lib-context` (https://www.npmjs.com/package/lit-context). Components will be able to subscribe to individual  `hass` property. Also, we pass the 3 parameters in `computeStateDisplay`.

I'm ok to pass the whole `hass` object for now but we will need to find a solution for the future context feature.
May be we can expose `computeStateDisplay` and `computeAttributeValueDisplay` the same way we expose `localize` (but that's not the scope of this PR)

Reply:
I see.

I'm happy to adjust is and pass in all three instead @piitaya

Reply:
Done in [360b3a1](https://github.com/home-assistant/frontend/pull/15822/commits/360b3a1c73155c03a52022110b0f8dbf82dda590)